### ISSUE TITLE
Search for hashtags doesn't return any result

### ISSUE THREAD

**Issue by fenarinarsa** [bluesky-social/social-app#2491]
**Describe the bug**

When searching for any hashtag, the search returns nothing.

**To Reproduce**

1. Search for a hashtag for example #hazbinhotel

**Expected behavior**

The search should return posts containing this hashtag.

**Details**

 - Platform: web and iOS
 - App version: 1.63

**Additional context**

While still not supported as interactive elements, hashtags are widely used for people and feeds to tag and find contents (especially for fandoms, series, artists). Sometimes hashtags are the only way to find content since some shows/topics use everyday words that would otherwise return non-relevant content.

**Comment by mschwendener** [comment]
I run into this a few days ago. Am 99,9% sure that it broke somewhere along the way and used to work in earlier versions.

**Comment by mschwendener** [comment]
> not like the previous one actually searched by hashtag IIRC

Yes, I seem to be mistaken. From [June 2023](https://github.com/bluesky-social/social-app/issues/848): "Even in its simplest form (allow search for them ..."

**Comment by fenarinarsa** [comment]
Another issue is that when a word is into a hashtag, you can't even look for it.

A post with #house won't show up in search results if you look for "house".

**Comment by HeyItsLollie** [comment]
This seems to have been somewhat improved — You can now search for "art", and bluesky will return posts containing both "art" and "#art". However, searching for "#art" results in the term being ignored. The easiest way to see this in action: Searching "from:username.bsky.app #art" will unexpectedly return all posts by the user, whereas "from:username.bsky.app art" will act as expected and only return posts from the user containing the word "art".

**Comment by kisaragi-hiu** [comment]
I think I know why hashtags aren't shown.

The search is performed in `Search.tsx`, via `useActorSearch` in `actor-search.ts`, which calls `searchActors` on the agent. The agent is defined by atproto/api; the [`searchActors` method](https://github.com/bluesky-social/atproto/blob/5f9ff1f17fd699b1c0f7944171e1d0e033c40332/packages/api/src/client/index.ts#L1373-L1382)) does an API call to `app.bsky.actor.searchActors`. The query eventually passes through [the `parseQuery` function in the backend](https://github.com/bluesky-social/indigo/blob/a30de45cd80176c16d826108914e8700d9ece8fd/search/parse_query.go#L16).

That `parseQuery` splits the query using https://github.com/google/shlex, but shlex (as the name implies) splits the string as if it's shell syntax, throwing away shell comments\* - everything after and including "#". This causes the query "#a_tag" to be equivalent to "" (empty string), and `from:username.bsky.app #art` to be equivalent to `from:username.bsky.app`.

\*This behavior is easily reproduced on [the Go Playground](https://go.dev/play/):

```go
package main

import (
	"fmt"
	"github.com/google/shlex"
)

func main() {
	// [abc foo]
	fmt.Println(shlex.Split("abc foo #def ghi"))
}
```

---

Additionally, this:

https://github.com/bluesky-social/social-app/blob/b9301befb701e3562de81294048163c62349bcbe/src/state/queries/actor-search.ts#L14-L16

- `prefix` probably needs to be url-encoded?
- I'm guessing `term` should be replaced with `q`, as [`term` is marked as deprecated in atproto/api](https://github.com/bluesky-social/atproto/blob/5f9ff1f17fd699b1c0f7944171e1d0e033c40332/packages/api/src/client/types/app/bsky/actor/searchActors.ts#L11-L18)

**Comment by Zero3K** [comment]
A fix has been made as a PR at https://github.com/bluesky-social/indigo/pull/579.

**Comment by Zero3K** [comment]
It has been merged. I think it'll take effect when a new version of Bluesky is uploaded to the servers.
